targets/spidermonkey/templates/register.c

```diff
@@ -16,7 +16,15 @@ void js_${generator.prefix}_${current_class.class_name}_finalize(JSFreeOp *fop,
 #if $generator.script_control_cpp
     js_proxy_t* nproxy;
     js_proxy_t* jsproxy;
+
+\#if (COCOS2D_VERSION >= 0x00031000)
+    JSContext *cx = ScriptingCore::getInstance()->getGlobalContext();
+    JS::RootedObject jsobj(cx, obj);
+    jsproxy = jsb_get_js_proxy(jsobj);
```

Comment:
this change wouldn't be necessary. the API for `jsb_get_js_proxy()` is still the same as v3.9

```c++
// v3.10 API:
js_proxy_t* jsb_get_js_proxy(JSObject* jsObj);
```